### PR TITLE
add: added heatlh check endpoint

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

Added a health endpoint to allow external services to confirm backend readiness before initiating dependent services.

Utilizing this endpoint ensures frontend e2e tests via Cypress run only after backend is fully operational, instead of relying on `sleep 5 `command.

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -204,6 +204,18 @@ jobs:
           path: frontend
           fetch-depth: 1
 
+      - name: Wait for Backend
+        run: |
+          npm install -g wait-on && \
+          wait-on http://localhost:3000/api/v3/health --timeout 200000
+        # run: |
```
Comment (martin-trajanovski) [inline .github/workflows/test.yml]: Maybe the commented part should be removed completely.

Comment (Junjiequan) [reply .github/workflows/test.yml:211]: Yes I will remove them, I need to re-run this test more times to see if it is stable

Comment (Junjiequan) [reply .github/workflows/test.yml]: Yes I will remove them, I need to re-run this test more times to see if it is stable

**Review by martin-trajanovski — APPROVED** [review]
This looks good to me now

**Review by nitrosx — APPROVED** [review]

**Pull request merged**